### PR TITLE
Fixes #30863: support update source_address_prefix in azure_rm_securitygroup

### LINKED ISSUE #30863: Try to update azure_rm_securitygroup : source_address_prefix
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

`azure_rm_securitygroup`

##### ANSIBLE VERSION

```
2.4.0
```

##### OS / ENVIRONMENT

CentOS 7.3

Steps of the Ansible installation
```
sudo yum check-update; sudo yum install -y gcc libffi-devel python-devel openssl-devel epel-release
sudo yum install -y python-pip python-wheel

sudo pip install ansible
sudo pip install ansible[azure]
sudo pip install dnspython
```

##### SUMMARY

Hello,

When I try to update network security group rules for Azure, in fact the option **source_address_prefix**, the value updated in the YML is not updated in Azure. However option like destination_port_range are updated when modified

##### STEPS TO REPRODUCE

Original YML

```yaml

  - name: Create security group that allows SSH and HTTP
    azure_rm_securitygroup: 
      resource_group: '{{ resourceGroup }}'
      name: 'NSG-{{ nsgName }}'
      purge_rules: yes
      state: present
      rules: 
        - name: SSH-Ansible
          protocol: Tcp
          destination_port_range: 22
          access: Allow
          priority: 100
          direction: Inbound
          source_address_prefix: 1.1.1.1


```

Modified YML

```yaml

  - name: Create security group that allows SSH and HTTP
    azure_rm_securitygroup: 
      resource_group: '{{ resourceGroup }}'
      name: 'NSG-{{ nsgName }}'
      purge_rules: yes
      state: present
      rules: 
        - name: SSH-Ansible
          protocol: Tcp
          destination_port_range: 22
          access: Allow
          priority: 100
          direction: Inbound
          source_address_prefix: 2.2.2.2


```
_strings in {{ }} are parameters passed with --extra-vars_

##### EXPECTED RESULTS

The IP source has to be updated in Azure.

##### ACTUAL RESULTS

Nothing is updated

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

Fixes #30863
Fix the security group cannot update source data prefix problem

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

azure_rm_securitygroup

##### ANSIBLE VERSION

```
2.4
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/integration/targets/azure_rm_securitygroup/tasks/main.yml [hunk 1]
```diff
@@ -41,30 +41,35 @@
       resource_group: "{{ resource_group }}"
       name: mysecgroup
       rules:
-          - name: DenySSH
+          - name: 'AllowSSH'
```
Comment (xscript) [inline test/integration/targets/azure_rm_securitygroup/tasks/main.yml]: quote is unnecessary

File: test/integration/targets/azure_rm_securitygroup/tasks/main.yml [hunk 2]
```diff
@@ -41,30 +41,35 @@
...
+          - "{{ output.state.rules | length }} == 3"
+          - output.state.rules[1].source_address_prefix == '174.108.158.0/24'
 
 - name: Test idempotence
   azure_rm_securitygroup:
       resource_group: "{{ resource_group }}"
       name: mysecgroup
       rules:
-          - name: DenySSH
+          - name: 'AllowSSH'
```
Comment (xscript) [inline test/integration/targets/azure_rm_securitygroup/tasks/main.yml]: remove quote

**Comment by xscript** [comment]
shipit

**Comment by yaweiw** [comment]
shipit

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
rebuild_merge

**Pull request merged**